### ISSUE TITLE
segmentation fault  swiftlint lint --fix --format

### ISSUE THREAD

**Issue by Adobels** [realm/SwiftLint#5365]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

When @available is followed by #Preview() {}, Swiftlint with fix and format flags prints this error 

```
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.
Correcting 'File.swift' (26/44)
zsh: segmentation fault  swiftlint lint --fix --format
```

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint --fix --format
```

```
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.
Correcting 'File.swift' (26/44)
zsh: segmentation fault  swiftlint lint --fix --format
```

### Environment

* SwiftLint version 0.54.0
* Installation method used: building from source
* Paste your configuration file:

```yml
disabled_rules:
  - type_name
  - line_length
  - identifier_name
  - trailing_comma
opt_in_rules:
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? NO
* Which Xcode version are you using (check `xcodebuild -version`)? Xcode 15.1.Beta3

**Sample:**

```swift
// This triggers a error:
@available(iOS 17, *)
#Preview("") {
    UIView()
}
```

```swift
// This doesn't triggers a error:
#Preview("") {
    UIView()
}
```

```bash
$ swiftlint lint --fix --format
```

**Comment by SimplyDanny** [comment]
Please try the latest version 0.54.0. Could be that the parser in 0.51.0 just wasn't able to read such code.

**Comment by Adobels** [comment]
Thank you for the suggestion. I have updated SwiftLint to version 0.54.0, but I'm still encountering the same issue. I have also updated the issue with this information that the issue is present under the 0.54.0 version.

**Comment by jpsim** [comment]
If the format flag is necessary to trigger the crash then this is likely a SourceKit bug, since format just tells SourceKit to format the file.

**Comment by Adobels** [comment]
@jpsim Thanks for your quick response. I'm not sure whether I should create the issue in the SourceKit project or if you'd prefer to handle it since you're more familiar with SwiftLint's dependencies. What do you think would be the best way to proceed?

**Comment by jpsim** [comment]
Reported this here: https://github.com/apple/swift/issues/70002

Will close this issue since the crash is happening in the Swift project itself and there's nothing to do in SwiftLint.

**Comment by krishsatya** [comment]
@Adobels I've just run into this same crash. Were you able to come up with a temporary workaround? I suppose I can move the previews into a separate source file that is ignored by swiftlint.

**Comment by Adobels** [comment]
This issue was addressed in https://github.com/swiftlang/swift/issues/70002. Soon, the toolchains and the project will be updated to address this issue

**Comment by Adobels** [comment]
> @Adobels I've just run into this same crash. Were you able to come up with a temporary workaround? I suppose I can move the previews into a separate source file that is ignored by swiftlint.

We have removed -format from the command and want to reactivate it after the issue is fixed.
